File: StraviaTec-Web/Models/CategoriaCarrera.cs
using System;
using System.Collections.Generic;

namespace StraviaTec_Web.Models
{
    public partial class CategoriaCarrera
    {
        public CategoriaCarrera()
        {
            InscripcionEvento = new HashSet<InscripcionEvento>();
        }

        public int Id { get; set; }
        public int IdCarrera { get; set; }
        public int IdCategoria { get; set; }

        public virtual Carrera IdCarreraNavigation { get; set; }
        public virtual Categoria IdCategoriaNavigation { get; set; }
        public virtual ICollection<InscripcionEvento> InscripcionEvento { get; set; }
    }
}

File: StraviaTec-Web/ClientApp/src/app/models/categoriaCarrera.ts
export class CategoriaCarrera {
    id: number;
    idCarrera: number;
    idCategoria: number;
    nombre: string;

    constructor(
        id: number, 
        idCarrera: number, 
        idCategoria: number) 
        {
            this.id = id;
            this.idCarrera = idCarrera;
            this.idCategoria = idCategoria;
        }
}